autoload/go/term.vim

```diff
@@ -30,49 +30,73 @@ function! go#term#newmode(bang, cmd, errorformat, mode) abort
-  setlocal bufhidden=delete
-  setlocal winfixheight
-  setlocal noswapfile
-  setlocal nobuflisted
-
-  " explicitly bind callbacks to state so that within them, self will always
-  " refer to state. See :help Partial for more information.
-  "
-  " Don't set an on_stderr, because it will be passed the same data as
-  " on_stdout. See https://github.com/neovim/neovim/issues/2836
-  let l:job = {
-        \ 'on_stdout': function('s:on_stdout', [], state),
-        \ 'on_exit' : function('s:on_exit', [], state),
-      \ }
-
-  let l:state.id = termopen(a:cmd, l:job)
-  let l:state.termwinid = win_getid(winnr())
-
-  execute l:cd . fnameescape(l:dir)
-
-  " resize new term if needed.
   let l:height = go#config#TermHeight()
   let l:width = go#config#TermWidth()
 
-  " Adjust the window width or height depending on whether it's a vertical or
-  " horizontal split.
-  if l:mode =~ "vertical" || l:mode =~ "vsplit" || l:mode =~ "vnew"
-    exe 'vertical resize ' . l:width
-  elseif mode =~ "split" || mode =~ "new"
-    exe 'resize ' . l:height
-  endif
+  " setup job for nvim
+  if has('nvim')
+    execute l:mode . ' __go_term__'
+    setlocal filetype=goterm
+    setlocal bufhidden=delete
+    setlocal winfixheight
+    setlocal noswapfile
+    setlocal nobuflisted
+
+    " explicitly bind callbacks to state so that within them, self will always
+    " refer to state. See :help Partial for more information.
+    "
+    " Don't set an on_stderr, because it will be passed the same data as
+    " on_stdout. See https://github.com/neovim/neovim/issues/2836
+    let l:job = {
+          \ 'on_stdout': function('s:on_stdout', [], state),
+          \ 'on_exit' : function('s:on_exit', [], state),
+        \ }
+    let l:state.id = termopen(a:cmd, l:job)
+    let l:state.termwinid = win_getid(winnr())
+    execute l:cd . fnameescape(l:dir)
+
+    " Adjust the window width or height depending on whether it's a vertical or
+    " horizontal split.
+    if l:mode =~ "vertical" || l:mode =~ "vsplit" || l:mode =~ "vnew"
+      exe 'vertical resize ' . l:width
+    elseif mode =~ "split" || mode =~ "new"
+      exe 'resize ' . l:height
+    endif
+    " we also need to resize the pty, so there you go...
+    call jobresize(l:state.id, l:width, l:height)
+
+  " setup term for vim8
+  elseif has('terminal')
+    let l:term = {
+          \ 'term_rows' : l:height,
+          \ 'term_cols' : l:width,
+          \ 'out_cb': function('s:out_cb', [], state),
+          \ 'exit_cb' : function('s:exit_cb', [], state),
+        \ }
+
+    if l:mode =~ "vertical" || l:mode =~ "vsplit" || l:mode =~ "vnew"
+      let l:term["vertical"] = 1
+    endif
 
-  " we also need to resize the pty, so there you go...
-  call jobresize(l:state.id, l:width, l:height)
+    let l:id = term_start(["/bin/sh", "-c", a:cmd], l:term)
+    let l:state.id = l:id
+    let l:state.termwinid = win_getid(bufwinnr(l:id))
+    execute l:cd . fnameescape(l:dir)
+  else 
+    return -1
```

Comment:
Given that `go#config#TermMode` is checked in every instance before calling this function and that `go#config#TermMode` checks that terminal mode is supported, is this else needed?

Reply:
You are correct. Will remove.